connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -185,46 +180,105 @@ public Task newTask(Class<? extends Task> taskClass) {
         return newPlugin(taskClass);
     }
 
-    public Converter newConverter(String converterClassOrAlias) {
-        return newConverter(converterClassOrAlias, null);
+    /**
+     * Get an instance of the give class specified by the given configuration key.
+     *
+     * @param key The configuration key for the class
+     * @param t The interface the class should implement
+     * @return A instance of the class
+     */
+    private <T> T getInstance(AbstractConfig config, String key, Class<T> t) {
+        Class<?> c = config.getClass(key);
+        if (c == null)
+            return null;
+        Object o = Utils.newInstance(c);
+        if (!t.isInstance(o))
```

Comment:
Shouldn't this be in the contract of `Utils.newInstance` to not return some other class that doesn't match?

I think this is pulled from `AbstractConfig` which makes sense for consistency, but I don't get why `Utils.newInstance` would ever return a value with an invalid type.

Reply:
The problem is that the class we found in the configuration, `c`, might not actually implement the type `t` that was supplied in the params. We have to do that manually.

BTW, we have to replicate a lot of this logic because `AbstractConfig` has a method to return a _configured_ instance, but no method to return a _non-configured_ instance. The proper way to address this is to add that method to `AbstractConfig`, but that requires a KIP, right?